ui/tests/unit/abilities/variable-test.js

```diff
@@ -1069,5 +1069,76 @@ module('Unit | Ability | variable', function (hooks) {
         'It should return the exact path match.'
       );
     });
+
+    test('it handles globs in namespaces', function (assert) {
+      const mockToken = Service.extend({
+        aclEnabled: true,
+        selfToken: { type: 'client' },
+        selfTokenPolicies: [
+          {
+            rulesJSON: {
+              Namespaces: [
+                {
+                  Name: '*',
```

Comment:
Given the example in https://github.com/hashicorp/nomad/issues/14034 should we test against a partial match example? ex. include a rule for `pab*` namespace as well, so that we verify we're getting the nearest-match behavior we'd expect.

Reply:
I agree with that sentiment and this is where I wonder how far to take testing.

That logic is handled by another [function](https://github.com/hashicorp/nomad/blob/main/ui/app/abilities/abstract.js#L83-L126) that is used throughout the application - but not tested.

So should we write unit tests for that other function because that functionality is encapsulated within that separate function we're invoking.

No one ever taught me functional programming and I feel like I need more clarity on cases like this.

Reply:
The way I look at it is that you want to figure out the right balance of coverage vs cost (both runtime and cost for you to crank out the tests) . 

The nice thing about testing the `_findMatchingNamespace` directly is that there's less setup involved in the test so you can test a larger number of combinations more cheaply to get full coverage of that function, and then you know any callers can rely on it being correct. Then you can potentially cover the callers (and their callers, etc.) at a higher level of abstraction. In the case of this test, the only behavior you're exercising is that of `_findMatchingNamespace`, so that suggests that might be a better place to put the test.